test/e2e/declarative_test_fwk/helper.py

```diff
@@ -186,13 +201,32 @@ def _assert_conditions(self, expectations: dict, reference: k8s.CustomResourceRe
             else:
                 raise Exception(f"Condition {condition_name} is provided with invalid value: {expected_value} ")
 
-    def assert_items(self, expectations: dict, state: dict) -> None:
-        """Asserts state against supplied expectations
+    def assert_items_aws(self, expectations: dict, actual: dict) -> None:
```

Comment:
I would suggest changing `actual` to just `spec` here. We use the terms `expect` and `actual` to refer to the things being compared. But this method doesn't compare `actual`, it compares the boto3 response.

Reply:
It not only compares `spec` but also compares `status`. API calls from boto3 response return fields from both `status` and `spec`. It doesn't really have keywords called `status` or `spec` from boto3 response.

Reply:
Changed all `actual` to `k8s_resource` to make code more readable.

Reply:
```
  def assert_extra_items_aws(self, expected_name, expected_value, aws_resource) -> bool:
        """Asserts specific fields that boto3 response against supplied expectation
        Override it as needed for validations of specific field
        If resource has expected_name as a field, execute custom logic to validate that specific field and return true
        Otherwise, if resource doesn't have expected_name as a field, return false
        Args:
            expected_name: expected name of a specific field to assert in aws resource
            expected_value: expected value of a specific field to assert in aws resource
            aws_resource: specific resource from aws memorydb
        Returns:
            True/False
        """
        return False
```